### PR TITLE
Sync annot-tuple-members with master

### PR DESCRIPTION
**Pull request by ushirask**

### CODE REVIEW

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/diagnostics/DiagnosticErrorCode.java [hunk 1]
```diff
@@ -351,11 +354,9 @@ public enum DiagnosticErrorCode implements DiagnosticCode {
     RESOURCE_ACCESS_SEGMENT_IS_NOT_ALLOWED_AFTER_REST_SEGMENT("BCE0680",
             "error.resource.access.segment.is.not.allowed.after.rest.segment"),
     ERROR_INVALID_TOKEN_IN_REG_EXP("BCE0681", "error.invalid.token.in.reg.exp"),
-    ERROR_INVALID_UNICODE_PROP_ESCAPE_IN_REG_EXP("BCE0682", "error.invalid.unicode.property.escape.in.reg.exp"),
-    ERROR_INVALID_QUANTIFIER_IN_REG_EXP("BCE0683", "error.invalid.quantifier.in.reg.exp"),
-    ERROR_INVALID_CHARACTER_SET_IN_REG_EXP("BCE0684", "error.invalid.character.set.in.reg.exp"),
-    ERROR_INVALID_FLAG_IN_REG_EXP("BCE0685", "error.invalid.flag.in.reg.exp"),
-    ERROR_ANNOTATIONS_NOT_ALLOWED_FOR_TUPLE_REST_DESCRIPTOR("BCE0686",
+    ERROR_INVALID_FLAG_IN_REG_EXP("BCE0682", "error.invalid.flag.in.reg.exp"),
+    ERROR_INVALID_QUANTIFIER_IN_REG_EXP("BCE0683", "error.invalid.quantifier.in.reg.exp")
```
Comment (MaryamZi) [inline compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/diagnostics/DiagnosticErrorCode.java:358]:
```suggestion
    ERROR_INVALID_QUANTIFIER_IN_REG_EXP("BCE0683", "error.invalid.quantifier.in.reg.exp"),
```

Please check the build and make sure changes are valid.

File: compiler/ballerina-parser/src/main/resources/syntax_diagnostic_message.properties [hunk 2]
```diff
@@ -334,8 +337,6 @@ error.rest.parameter.cannot.be.included.record.parameter=rest parameter cannot b
 error.resource.access.segment.is.not.allowed.after.rest.segment=\
   resource access path segment is not allowed after resource access rest segment
 error.invalid.token.in.reg.exp=invalid token in regular expression
-error.invalid.unicode.property.escape.in.reg.exp=invalid unicode property escape in regular expression
-error.invalid.quantifier.in.reg.exp=invalid quantifier in regular expression
-error.invalid.character.set.in.reg.exp=invalid character set in regular expression
 error.invalid.flag.in.reg.exp=invalid flag in regular expression
 error.annotations.not.allowed.for.tuple.rest.descriptor=annotations not allowed for tuple rest descriptor
+error.invalid.quantifier.in.reg.exp=invalid quantifier in regular expression
```
Comment (MaryamZi) [inline compiler/ballerina-parser/src/main/resources/syntax_diagnostic_message.properties:342]:
```suggestion
error.invalid.quantifier.in.reg.exp=invalid quantifier in regular expression

```

Comment (MaryamZi) [inline compiler/ballerina-parser/src/main/resources/syntax_diagnostic_message.properties:342]: Order isn't consistent with codes.

Comment (ushirask) [reply compiler/ballerina-parser/src/main/resources/syntax_diagnostic_message.properties]: Fixed,

File: compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/diagnostics/DiagnosticErrorCode.java (hunk 1 above)
Comment (MaryamZi) [reply compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/diagnostics/DiagnosticErrorCode.java:358]: Not fixed yet?

**Review by MaryamZi — APPROVED** [review]

**Pull request merged**